### PR TITLE
Switch to Doctrine naming conventions and coding standard

### PR DESCRIPTION
**Pull request by driehle**
This PR updates the project to better fit the Doctrine ecosphere:

- folder `test` is renamed to `tests`, as discussed in Slack
- coding standard is switched form laminas-coding-standard to doctrine/coding-standard
- two rules (`*MissingNativeTypeHint`) have been excluded as they'd introduce BC breaks
- three rules (`*MissingTraversableTypeHintSpecification`) have been excluded as they are currently too noisy
- one issue with `SuperfluousPrefix` in `AbstractCollectionStrategy` has been suppressed, as we cannot rename the class without having a BC break

Since I don't plan to release a new major soon, I think this is the best way to go for now.

### CODE REVIEW

File: phpcs.xml.dist [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
+<ruleset>
+    <arg name="basepath" value="."/>
+    <arg name="extensions" value="php"/>
+    <arg name="parallel" value="80"/>
+    <arg name="cache" value=".phpcs.cache"/>
+    <arg name="colors"/>
+
+    <!-- set minimal require PHP version (7.3) -->
```
Comment (greg0ire) [inline phpcs.xml.dist]:
```suggestion
    <!-- set minimal required PHP version (7.3) -->
```

File: phpcs.xml.dist [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <file>src</file>
+    <file>tests</file>
+
+    <!-- Include full Doctrine Coding Standard -->
+    <rule ref="Doctrine">
+        <!-- This causes BC breaks, as method signatures (return type) change -->
+        <exclude name="SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingNativeTypeHint"/>
+        <!-- This causes BC breaks, as method signatures (parameter type) change -->
+        <exclude name="SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint"/>
+        <!-- This is simply to noisy for now -->
```
Comment (greg0ire) [inline phpcs.xml.dist]:
```suggestion
        <!-- This is simply too noisy for now -->
```

File: phpcs.xml.dist [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        <exclude name="SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingTraversableTypeHintSpecification"/>
+        <exclude name="SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingTraversableTypeHintSpecification"/>
+        <exclude name="SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingTraversableTypeHintSpecification"/>
```
Comment (greg0ire) [inline phpcs.xml.dist:28]: Have you considered not excluding any rule and using a baseline instead? That way new code is forced to be up to the end goal.

Comment (driehle) [reply phpcs.xml.dist:28]: I know there is a baseline feature available and used for Psalm and PhpStan in the [Doctrine organization](https://github.com/search?p=1&q=org%3Adoctrine+baseline&type=Code), but for PhpCs? I searched [Laminas](https://github.com/search?p=1&q=org%3Alaminas+baseline&type=Code) as well, but again only Psalm and PhpStan. Is there really such feature for PhpCs?

Comment (greg0ire) [reply phpcs.xml.dist:28]: Oh sorry, that was me mixing things up!

**Review by greg0ire — APPROVED** [review]

**Review by TomHAnderson — APPROVED** [review]

**Pull request merged**